src/integration/java/io/pinecone/integration/dataPlane/UpsertAndDescribeIndexStatsTest.java

```diff
@@ -157,9 +157,12 @@ public void upsertOptionalVectorsAndQueryIndexFutureTest() throws InterruptedExc
                     true).get();
 
             ScoredVectorWithUnsignedIndices scoredVectorV1 = null;
-            for (int i = 0; i < topK; i++) {
-                if (upsertIds.get(0).equals(queryResponse.getMatches(i).getId())) {
-                    scoredVectorV1 = queryResponse.getMatches(i);
+            // if the sizes are not equal, let the following assertions fail and retry again
```

Comment:
Should have made this change as a part of a separate PR, sorry :( but if there's any latency in querying the vectors, the test will fail with IndexOutOfBounds exception so instead, if the lists sizes are not equal, let the assertions fail and assertWithRetry will handle it.